### PR TITLE
SC-458 Update Java adapter with new methods

### PR DESCRIPTION
**Pull request by xrtm000**

### CODE REVIEW

File: lang/jvm/src/main/java/com/wavesplatform/lang/Script.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package com.wavesplatform.lang;
+
+import com.wavesplatform.common.utils.Base64;
+
+public interface Script {
+    int version();
+
+    boolean isExpression();
```
Comment (msmolyakov) [inline lang/jvm/src/main/java/com/wavesplatform/lang/Script.java:8]: maybe single enum field?

File: lang/jvm/src/main/scala/com/wavesplatform/lang/JavaAdapter.scala [hunk 2]
```diff
@@ -1,29 +1,171 @@
...
+        directives   <- DirectiveParser(input)
+        directiveSet <- Directive.extractDirectives(directives, defaultStdLib = V4)
+        compile = if (directiveSet.contentType == DApp)
+          compileDApp _
+        else
+          compileExpression _
+        compiled <- compile(input, ctxCache(directiveSet).compilerContext, directiveSet)
+      } yield compiled
+
+    script.fold(error => throw new RideException(error), identity)
```
Comment (msmolyakov) [inline lang/jvm/src/main/scala/com/wavesplatform/lang/JavaAdapter.scala:52]: maybe handled exception?

**Review by msmolyakov — CHANGES_REQUESTED** [review]

File: lang/jvm/src/main/java/com/wavesplatform/lang/Script.java (hunk 1 above)
Comment (xrtm000) [reply lang/jvm/src/main/java/com/wavesplatform/lang/Script.java:8]: Why?
There are few options.

File: lang/jvm/src/main/scala/com/wavesplatform/lang/JavaAdapter.scala (hunk 2 above)
Comment (xrtm000) [reply lang/jvm/src/main/scala/com/wavesplatform/lang/JavaAdapter.scala:52]: Checked exception?
It is not so useful.

**Review by potan — APPROVED** [review]

**Review by mrkraft — APPROVED** [review]

**Pull request closed without merging**